src/main/java/build/buildfarm/instance/memory/MemoryLRUContentAddressableStorage.java

```diff
@@ -0,0 +1,156 @@
+package build.buildfarm.instance.memory;
+
+import build.buildfarm.common.ContentAddressableStorage;
+import build.buildfarm.common.ContentAddressableStorage.Blob;
+import build.buildfarm.common.Digests;
+import com.google.devtools.remoteexecution.v1test.Digest;
+import com.google.protobuf.ByteString;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class MemoryLRUContentAddressableStorage implements ContentAddressableStorage {
+  private final long maxSizeInBytes;
+  private final Map<Digest, Entry> storage;
```

Comment:
You're basically reimplementing LinkedHashMap. Did you know that LinkedHashMap supports being used as an LRU-style cache?

Reply:
It does, but it does not facilitate making the expiration decision based on an aggregate value (size in this case, I first tried to make it work as a subclass). There are also a couple of other optimizations that might not be obvious regarding the self-contained nature of the Blob and its keys.  This implementation was influenced heavily by the nature of LinkedHashMap.